### PR TITLE
HiFiC adversarial losses

### PR DESCRIPTION
**Pull request by 0x00b1**
- least squares adversarial loss
- non-saturating adversarial loss

### CODE REVIEW

**Comment by 0x00b1** [comment]
@mmuckley The two adversarial losses used in the HiFiC paper. I have mixed feelings about this PR. They are, of course, useful and certainly useful beyond a HiFiC implementation but I think they are out of scope for this package. Unfortunately, I don't know if there's a better place for them and I'm uncertain about the roadmap (if one exists) for general-purpose adversarial functionality in PyTorch. I suppose an alternative option worthy of consideration is moving them into `neuralcompression.models` and making them private, restricting their use to the HiFiC implementation.

**Comment by mmuckley** [comment]
@0x00b1 I think it makes perfect sense to make them private to HiFiC at the moment since it's just one model. If we have multiple models with discriminator losses then we can look at how to harmonize them.

**Comment by 0x00b1** [comment]
@mmuckley Great! I’ll go ahead and make those changes.

**Review by mmuckley — APPROVED** [review]
LGTM, although I couldn't find where they mention the least squares adversarial loss in the paper. Let me know if you want me to merge as-is.

File: neuralcompression/models/_hific/_least_squares_adversarial_loss.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import typing
```
Comment (mmuckley) [inline neuralcompression/models/_hific/_least_squares_adversarial_loss.py:1]: Please add a copyright header.
```
"""
Copyright (c) Facebook, Inc. and its affiliates.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
"""
```

File: neuralcompression/models/_hific/_non_saturating_adversarial_loss.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import typing
```
Comment (mmuckley) [inline neuralcompression/models/_hific/_non_saturating_adversarial_loss.py:1]: Please add a copyright header.
```
"""
Copyright (c) Facebook, Inc. and its affiliates.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
"""
```

**Review by mmuckley — CHANGES_REQUESTED** [review]

**Pull request merged**